q.js

```diff
@@ -1540,6 +1540,82 @@ Promise.prototype.all = function () {
 };
 
 /**
+ * Returns the first resolved promise of an array. Prior rejected promises are
+ * ignored.  Rejects only if all promises are rejected with an array
+ * containing the reasons of all rejected promises.
+ * @param {Array*} an array containing values or promises for values
+ * @returns a promise fulfilled with the value of the first resolved promise,
+ * or a promise rejected with an array with the reasons of all rejected
+ * promises.
+ */
+Q.firstResolved = firstResolved;
+
+function firstResolved(promises) {
+    var deferred = Q.defer();
+    if (promises.length === 0) {
+        deferred.resolve();
+    }
+
+    for (var i = 0; i < promises.length; i++) {
+        var promise = promises[i];
+        /* Let's define that false could be a value to resolve with, so let's not
+         * if(!promise)
+         */
+        if (promise === undefined || promise === null) {
+            continue;
+        }
+
+        if (!isPromiseAlike(promise)) {
+            // It's a value, not a promise
+            var value = promise;
+            deferred.resolve(value);
+            return deferred.promise;
+        }
+
+        (function(promise, i) {
+            promise.then(function(result) {
+                deferred.resolve(result);
+            }, function() {
+                // Ignore. Later, with allSettled, we take care of rejecting.
+            }, function(progress) {
+                deferred.notify({
+                    index: i,
+                    value: progress
+                });
+            });
+        })(promise, i);
+    }
+
+    var allRejected = [];
+
+    Q.allSettled(promises)
+      .then(function(allSettled) {
+        for (var i = 0; i < allSettled.length; i++) {
+            var snapshot = allSettled[i];
+            if (!snapshot || !snapshot.state) {
+              continue;
+            }
+            if (snapshot.state === 'fulfilled') {
+              return Q.reject();
+            } else if (snapshot.state === 'rejected') {
+              allRejected.push(Q.reject(snapshot.reason));
+            }
+        }
+
+        return Q.resolve(allRejected);
+      })
+      .then(function(result) {
+          deferred.reject(allRejected);
+      });
```

Comment:
Maybe it could be useful to know the reasons of all the rejected promises, for statistical purposes, or providing more information to users or while logging. Or for any other purpose it might be useful to know the individual reasons of all the rejected promises. What if I reject with an Error initialized with the array of reasons?: 
deferred.reject(new Error(allRejected)); instead of deferred.reject(allRejected);